connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -280,10 +282,7 @@ private void doCommit(Map<TopicPartition, OffsetAndMetadata> offsets, boolean cl
             OffsetCommitCallback cb = new OffsetCommitCallback() {
                 @Override
                 public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception error) {
-                    if (error == null) {
-                        lastCommittedOffsets = offsets;
-                    }
-                    onCommitCompleted(error, seqno);
+                    onCommitCompleted(error, seqno, offsets);
```

Comment:
This is the main invocation of `onCommitCompleted` that is called when the `consumer.commitAsync(...)` method finally calls our `OffsetCommitCallback`, and this change moves the setting of `lastCommittedOffsets` into the `onCommitCompleted` method.